developer-docs-site/docs/concepts/basics-gas-txn-fee.md

```diff
@@ -2,71 +2,122 @@
+  ```
+  (total gas units consumed) * (gas_price)
+  ```
+  must not exceed `max_gas_amount`, or else the transaction will abort the execution.
 
-## Using gas to prioritize a transaction
+The transaction fee charged to the client will be at the most `gas_price * max_gas_amount`.
 
-When you send a transaction, it is prioritized based on different criteria. One of these is the normalized gas price for the transaction.
+## Gas parameters set by the governance
 
-For example:
+The following gas parameters are set by the Aptos governance. 
 
-* Bob sends a transaction with `gas_price` 10.
-* Alice sends a transaction at the same time with `gas_price` 20.
+:::tip On-chain gas schedule
+These on-chain gas parameters are published on the Aptos blockchain at `0x1::gas_schedule::GasScheduleV2`.
+:::
 
-Alice's transaction would be ranked higher than Bob's.
+- `lower bound` : The `max_gas_amount` value in the transaction must be set to significantly greater than this parameter’s value.
+- `upper bound` : The maximum allowed value for the `max_gas_amount` gas parameter.
 
-## Core design principles
-Three central principles have motivated the design of gas in Aptos and Move:
+## Dynamic gas pricing for storage
 
-| Design Principle | Description |
-| ---------- | ---------- |
-| Move is Turing complete | Because of this, determining if a given Move program terminates cannot be decided statically. However, by ensuring that <br/>  - every bytecode instruction has a non-zero gas consumption, and <br/>  - the amount of gas that any program can be started with is bounded, <br/>  we get this termination property for programs almost free of cost. |
-| Discourage DDoS attacks and encourage judicious use of the network | The gas usage for a transaction is correlated with resource consumption of that transaction. The gas price, and hence the transaction fee, should rise-and-fall with contention in the network. At launch, we expect gas prices to be at or near zero. But in periods of high contention, you can prioritize transactions using the gas price, which will encourage sending only vital transactions during such times. |
-| The resource usage of a program needs to be agreed upon in consensus | This means that the method of accounting for resource consumption needs to be deterministic. This rules out other means of tracking resource usage, such as cycle counters or any type of timing-based methods as they are not guaranteed to be deterministic across nodes. The method for tracking resource usage needs to be abstract. |
+Aptos gas pricing uses dynamic prices for the storage operations. This means that the storage costs, hence the `gas_price`, can increase exponentially as the Aptos blockchain state database is filled up. The storage cost can become as high as `100x` at 100% utilization. However, the expectation is that the validators will make use of larger and cheaper storage hardware to mitigate such exponential rise in the storage costs. 
```

Comment:
This is the `gas_used` not the `gas_price` that increases.

The `gas_price` is determined by the market

The `gas_used` is determined by the resources